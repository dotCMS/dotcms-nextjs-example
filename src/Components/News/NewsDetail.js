import React from 'react';
import PropTypes from 'prop-types';
import DateFormat from '../Shared/DateFormat';

const NewsDetail = ({ title, sysPublishDate, image, story }) => {
    return (
        <>
            <h3>{title}</h3>
            <h5>
                <DateFormat dateString={sysPublishDate} />
            </h5>
            {image ? <img className="news-detail-img" alt={title} src={image} /> : ''}
            <div dangerouslySetInnerHTML={{ __html: story }} />
        </>
    );
};

NewsDetail.propTypes = {
    image: PropTypes.string,
    sysPublishDate: PropTypes.string,
    title: PropTypes.string,
    story: PropTypes.string
};

NewsDetail.defaultProps = {
    image: '',
    sysPublishDate: '',
    title: '',
    story: ''
};

export default NewsDetail;
