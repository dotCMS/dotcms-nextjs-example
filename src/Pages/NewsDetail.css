.news-detail__image {
    box-shadow: var(--card-shadow);
    margin: -75px 0 2rem 0;
    position: relative;
    width: 100%;
}