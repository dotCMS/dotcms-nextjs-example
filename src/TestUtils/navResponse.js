export const NavResponse = {
    entity: {
        children: [
            {
                href: '/about',
                title: 'About Us',
                folder: '123',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/products',
                title: 'Products',
                folder: '456',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/resources',
                title: 'Resources',
                folder: '789',
                type: 'folder',
                children: [
                    {
                        href: '/videos',
                        title: 'Videos',
                        type: 'htmlpage',
                        folder: null
                    },
                    {
                        href: '/blog',
                        title: 'Blog',
                        type: 'htmlpage',
                        folder: null
                    }
                ]
            }
        ]
    }
};
