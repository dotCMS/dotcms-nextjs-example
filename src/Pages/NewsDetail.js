import React from 'react';
import DotCMSApi from '../libs/dotcms.api';
import Layout from '../Components/Layout';
import NewsDetail from '../Components/News/NewsDetail';
import './NewsDetail.css';

class NewsDetailPage extends React.Component {
    constructor(props, context) {
        super(props, context);
        const { news } = props.location.state || {};
        this.state = {
            news: news
        };
    }

    getNews() {
        DotCMSApi.esSearch('news', {
            detailedSearchQuery: `+News.urlTitle:${this.props.match.params.slug}`
        })
            .then(response => response.json())
            .then(newsData => {
                this.setState(state => ({
                    ...state,
                    news: newsData.contentlets[0]
                }));
            });
    }

    componentDidMount() {
        if (!this.state.news) {
            this.getNews();
        }
    }

    render() {
        return (
            <Layout>
                <NewsDetail news={this.state.news} />
            </Layout>
        );
    }
}

export default NewsDetailPage;
