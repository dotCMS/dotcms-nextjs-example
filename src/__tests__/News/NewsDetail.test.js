import React from 'react';
import { shallow } from 'enzyme';
import NewsDetail from '../../Components/News/NewsDetail';
import toJSON from 'enzyme-to-json';

import { newsMock } from '../../TestUtils/news';

describe('<NewsDetail />', () => {
    it('renders NewsDetail correctly', () => {
        const wrapper = shallow(
            <NewsDetail news={newsMock} />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
