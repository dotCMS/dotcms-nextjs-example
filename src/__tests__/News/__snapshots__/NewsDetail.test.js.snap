// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsDetail /> renders NewsDetail correctly 1`] = `
<Fragment>
  <h3 />
  <h5 />
  <img
    alt=""
    className="news-detail-img"
    src=""
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</Fragment>
`;
